src/aosm/azext_aosm/build_processors/base_processor.py

```diff
@@ -185,11 +185,18 @@ def generate_values_mappings(
         # Loop through each property in the schema.
         for subschema_name, subschema in schema["properties"].items():
 
-            if "type" not in subschema:
-                if ["oneOf", "anyOf"] in subschema:
+            if isinstance(subschema, dict) and "type" not in subschema:
+                if "oneOf" in subschema:
                     raise InvalidArgumentValueError(
                         f"The subschema '{subschema_name}' does not contain a type.\n"
-                        "It contains 'anyOf' or 'oneOf' logic, which is not valid for AOSM.\n"
+                        "It contains 'oneOf' logic, which is not valid for AOSM.\n"
+                        "Please remove this from your values.schema.json and provide a concrete type "
+                        "or remove the schema and the CLI will generate a generic schema."
+                    )
+                if "anyOf" in subschema:
+                    raise InvalidArgumentValueError(
+                        f"The subschema '{subschema_name}' does not contain a type.\n"
+                        "It contains 'anyOf' logic, which is not valid for AOSM.\n"

```

Comment:
A small point, but I think having a single exception for both oneOf and anyOf is preferable. We stop execution as soon as we hit this. If the error message has details of both, the user can remove all instances of oneOf and anyOf at once. With separate messages, the user won't realise they need to hit both, so will hit the exception again on the second time through (of course only if they have both). A small change, but from experience one that we know users find particularly frustrating.

```suggestion
                        "It contains 'anyOf' or 'oneOf' logic, which is not valid for AOSM.\n"
```